sdks/python/apache_beam/internal/apiclient.py

```diff
@@ -165,7 +163,7 @@ def __init__(self, packages, options, environment_version):
           dataflow.Package(
               location='%s/%s' % (
                   self.google_cloud_options.staging_location.replace(
-                      'gs:/', STORAGE_API_SERVICE),
```

Comment:
Are you sure following code doesn't have to be modified ? This seems to be doing opposite of what is mentioned in PR description.

     if not self.google_cloud_options.temp_location:
       logging.info('Defaulting to the staging_location as temp_location: %s',
                    self.google_cloud_options.staging_location)
       (self.google_cloud_options
        .temp_location) = self.google_cloud_options.staging_location

Update you mentioned seems to be only be doing an error check not changing default. Did you try running a job without specifying the staging location ?